### PR TITLE
allow ALTER for TEMPORARY tables

### PR DESCRIPTION
**Pull request by icuken**
Related to #56336. 

### Changelog category:
- Improvement

### Changelog entry:
Allow basic commands in ALTER for TEMPORARY tables.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 90f9844fc886bf296ea7c41caa91126e78b533db with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56892/90f9844fc886bf296ea7c41caa91126e78b533db/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6897118915">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56892/90f9844fc886bf296ea7c41caa91126e78b533db/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by jkartseva — COMMENTED** [review]
Please update the documentation:

https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/statements/alter/column.md

**Review by jkartseva — CHANGES_REQUESTED** [review]
Asking to update the documentation, otherwise LGTM.

**Comment by icuken** [comment]
Do I understand correctly that adding `[TEMPORARY]` to the syntax section will be enough?

File: src/Storages/StorageMemory.h [hunk 1]
```diff
@@ -76,6 +76,8 @@ friend class MemorySink;
     void backupData(BackupEntriesCollector & backup_entries_collector, const String & data_path_in_backup, const std::optional<ASTs> & partitions) override;
     void restoreDataFromBackup(RestorerFromBackup & restorer, const String & data_path_in_backup, const std::optional<ASTs> & partitions) override;
 
+    void checkAlterIsPossible(const AlterCommands & commands, ContextPtr local_context) const override;
```
[the remark this answers is not in the document]
Comment (icuken) [reply src/Storages/StorageMemory.h:79]: It'll work on any temporary table, other engines will support all operations allowed for ALTER on non-temporary tables.

[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Storages/StorageMemory.h:79]: Could you please add a test with some other engine as well?

**Review by jkartseva — APPROVED** [review]
LGTM, thank you.

**Comment by jkartseva** [comment]
@icuken 
Could you please rebase for the test pass?

**Pull request merged**